src/span.js

```diff
@@ -287,13 +286,18 @@ export default class Span {
    */
   _setSamplingPriority(priority: number): boolean {
     this._spanContext.finalizeSampling();
-    if (priority == 0) {
-      this._spanContext.flags = this._spanContext.flags & ~constants.SAMPLED_MASK;
-      return true;
-    } else if (this._tracer._isDebugAllowed(this._operationName)) {
-      this._spanContext.flags = this._spanContext.flags | constants.SAMPLED_MASK | constants.DEBUG_MASK;
-      return true;
+    if (priority > 0) {
+      if (this._spanContext.isDebug()) {
+        // If the span is already in debug, no need to set it again
+        return false;
+      }
+      if (this._tracer._isDebugAllowed(this._operationName)) {
+        this._spanContext.flags = this._spanContext.flags | constants.SAMPLED_MASK | constants.DEBUG_MASK;
+        return true;
+      }
+      return false;
     }
+    this._spanContext.flags = this._spanContext.flags & ~constants.SAMPLED_MASK;
```

Comment:
Here you're updating the flag but returning false...?

Reply:
I made this change on purpose but now that you mention it, semantically it doesn't make sense. Even though this returning true or false has no impact on the span (since we're disabling sampling so the flag being set or not will never be surfaced to the user). I'll revert